tests/index.test.ts

```diff
@@ -535,56 +538,139 @@ describe('PTY', { repeats: 500 }, () => {
 });
 
 type CgroupState = {
-  sliceDir: string;
-  originalCgroup: string;
-  sliceName: string;
+  version: 'v1' | 'v2';
+  sliceDir: string; // For v2: Full path to the slice dir. For v1: Base name for the group (e.g., "ruspty-xyz").
+  originalCgroup?: string; // For v2: Full path to original cgroup dir. For v1: Path within each hierarchy (e.g., "/user.slice").
+  sliceName: string; // For v2: the slice file name. For v1: Base name for the group (same as sliceDir).
   moved: boolean;
+  v1Subsystems?: string[]; // Only for v1: List of subsystems to manage (e.g., ['cpu', 'memory']).
 };
```

Comment:
```suggestion
type CgroupState = {
  version: 'v1';
   // Path within each hierarchy (e.g., "/user.slice").
  originalCgroup: string;
  // Base name for the group.
  sliceName: string;
  moved: boolean;
  // List of subsystems to manage (e.g., ['cpu', 'memory']).
  v1Subsystems: string[];
} | {
  version: 'v2';
   // Full path to the slice dir.
  sliceDir: string;
   // Full path to original cgroup dir.
  originalCgroup: string;
   // The slice file name.
  sliceName: string;
  moved: boolean;
};
```

we have the technology to avoid having two types in superposition! (although see the above comment, might be simpler)